controller_manager/controller_manager/hardware_spawner.py

```diff
@@ -0,0 +1,229 @@
+            node, controller_manager_name, component, active_state, "activated"
+        )
+
+
+def configure_components(node, controller_manager_name, components_to_configure):
+    inactive_state = State()
+    inactive_state.id = State.PRIMARY_STATE_INACTIVE
+    inactive_state.label = "inactive"
+    for component in components_to_configure:
+        handle_set_component_state_service_call(
+            node, controller_manager_name, component, inactive_state, "configured"
+        )
+
+
+def main(args=None):
+    rclpy.init(args=args, signal_handler_options=SignalHandlerOptions.NO)
+    parser = argparse.ArgumentParser()
+    activate_or_confiigure_grp = parser.add_mutually_exclusive_group(required=True)
+
+    parser.add_argument(
+        "hardware_component_name",
+        help="The name of the hardware component which should be activated.",
+    )
+
+    parser.add_argument(
+        "-c",
+        "--controller-manager",
+        help="Name of the controller manager ROS node",
+        default="controller_manager",
+        required=False,
+    )
+    parser.add_argument(
+        "--controller-manager-timeout",
+        help="Time to wait for the controller manager",
+        required=False,
+        default=10,
+        type=int,
+    )
+
+    activate_or_confiigure_grp.add_argument(
+        "--activate",
+        help="Activates the given components. Note: Components are by default configured before activated. ",
+        action="store_true",
+        required=False,
+    )
+
+    activate_or_confiigure_grp.add_argument(
+        "--configure",
+        help="Configures the given components.",
+        action="store_true",
+        required=False,
+    )
+
+    command_line_args = rclpy.utilities.remove_ros_args(args=sys.argv)[1:]
+    args = parser.parse_args(command_line_args)
+    controller_manager_name = args.controller_manager
+    controller_manager_timeout = args.controller_manager_timeout
+    hardware_component = [args.hardware_component_name]
+    activate = args.activate
+    configure = args.configure
+
+    print(type(hardware_component))
+
+    node = Node("hardware_spawner")
+    if not controller_manager_name.startswith("/"):
+        spawner_namespace = node.get_namespace()
+        if spawner_namespace != "/":
+            controller_manager_name = f"{spawner_namespace}/{controller_manager_name}"
+        else:
+            controller_manager_name = f"/{controller_manager_name}"
+    try:
+        if not wait_for_controller_manager(
+            node, controller_manager_name, controller_manager_timeout
+        ):
+            node.get_logger().error("Controller manager not available")
+            return 1
+
+        if activate:
+            activate_components(node, controller_manager_name, hardware_component)
+        elif configure:
+            configure_components(node, controller_manager_name, hardware_component)
+        else:
+            node.get_logger().warn("Something went wrong.")
```

Comment:
This is rather an error.

Can you give out more useful output? When can this happen? If no flag is set, what will be default behavior? I propose to make “activate” the default action.
```suggestion
            node.get_logger().error("Something went wrong.")
```

Reply:
Technically should never happen since --activate and --configure are mutually exclusive and one is required.